flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -307,6 +312,119 @@ object UserDefinedFunctionUtils {
   // ----------------------------------------------------------------------------------------------
 
   /**
+    * get data view type information from accumulator constructor.
+    *
+    * @param aggFun aggregate function
+    * @return the data view specification
+    */
+  def getDataViewTypeInfoFromConstructor(
+    aggFun: AggregateFunction[_, _])
+  : mutable.HashMap[String, TypeInformation[_]] = {
+
+    val resultMap = new mutable.HashMap[String, TypeInformation[_]]
+    val acc = aggFun.createAccumulator()
+    val fields: util.List[Field] = TypeExtractor.getAllDeclaredFields(acc.getClass, true)
+    for (i <- 0 until fields.size()) {
+      val field = fields.get(i)
+      field.setAccessible(true)
+      if (classOf[DataView].isAssignableFrom(field.getType)) {
+        if (field.getType == classOf[MapView[_, _]]) {
+          val mapView = field.get(acc)
+          val keyTypeInfo = getFieldValue(classOf[MapView[_, _]], mapView, "keyTypeInfo")
```

Comment:
Add `private[flink]` to key and value type infos will make them public in Java. I'm not sure whether it is a good idea. I would like to not expose them to users (Java users), and the reflection only happens in compile which is fine I think.